build.sbt

```diff
@@ -414,6 +417,9 @@ lazy val scalaDottyBenchmarks = (project in file("benchmarks/scala-dotty"))
       // Version 3.1.2 was the last to compile with Scala 2.13.11. Version 3.1.3-RC2
       // broke compilation due to "Unsupported Scala 3 union in parameter value".
       // Compiling with Scala 3.1.0+ avoids compatibility issues.
+      // scala3-compiler_3 3.3.1 is the last dotty version to support JDK 21
+      // https://bugs.openjdk.org/browse/JDK-8308050
+      // https://github.com/scala/scala3/commit/f77069a74c4df65c7f69a6476aeb6b52fa8acb48
       "org.scala-lang" % "scala3-compiler_3" % "3.3.1",
```

Comment:
Perhaps this could be replaced with `scalaVersion3`. I think the explicit version is a leftover from the time we used dotty3 with Scala 2.13.

Reply:
OK, with respect to the above, we may need to keep `scala3-compiler_3` version separate from `scalaVersion3`. How odd.